### ISSUE TITLE
[Feature] Turn off Javascript

### ISSUE THREAD

**Issue by jjavierdguezas** [responsively-org/responsively-app#490]
Add buttons (per devices and for all) that toggle javascript in the device
This should be in the ` ⋮ ` button (_kebab menu_?) since it is not a usual action

**Hints:**
`<webview src="https://github.com" webpreferences="javascript=no"></webview>`
[`webpreferences`](https://www.electronjs.org/docs/api/webview-tag#webpreferences) is a comma separated list of strings, the full list of options can be found [here](https://www.electronjs.org/docs/api/browser-window#new-browserwindowoptions)

**Comment by smcveigh941** [comment]
Hey, @jjavierdguezas I'm looking at this :)

**Comment by jjavierdguezas** [comment]
Great @smcveigh941!
Assigned this to you

**Comment by manojVivek** [comment]
> I'm looking at this :)

@smcveigh941 Please join our slack if you have questions to discuss: https://responsively.app/join-slack

**Comment by ParamBirje** [comment]
Hey @manojVivek !

I added a button to toggle Javascript for each view. 
![js-toggle](https://github.com/responsively-org/responsively-app/assets/87022870/8b316d3e-8477-4a63-9e50-362e1f2c777f)

As suggested by @jjavierdguezas , i was fiddling around with the webviews trying this,
```
const toggleJavascript = () => {
    if (webview) {
      setAllowedJavascript((prev) => !prev);
      if (allowedJavascript) {
        webview.setAttribute('webpreferences', 'javascript=no');
      } else {
        webview.setAttribute('webpreferences', 'javascript=yes');
      }
      refreshView();
      console.log('attr', webview);
    }
  };
```
Though this **does** change the attribute status in the webview element, in the app, the webviews don't reflect this change *(they still allow js)*

But then i found `Previewer/Device/index.tsx`, where the webviews are created and added `webpreferences="javascript=off"` explicitly. 

And then the webviews **do** turn off JS in them with a catch in it, that ESLint gives an error as unknown attribute `webpreferences` *(which could be solved by ignoring that line in `react/no-unknown-property`)* but then above toggle function on this change still doesn't work.

Could you please guide me towards any necessary pointers to solve this?

cc: @themohammadsa

**Comment by manojVivek** [comment]
The reason, I'm guessing, could be the fact that when the attribute is added/changed after the webview instance is already created, it doesn't have any effect. 
If the instance is created with the attributes set, then it works in that mode irrespective of modifying the state of the attribute after the creation. 

A (hacky) solution that I can think of is to make sure the webview instance is destroyed and re-created whenever this option is changed.

**Comment by ParamBirje** [comment]
I have implemented the solution as per @manojVivek 's pointers and it does successfully work. 

![allow-js-kebab-menu](https://github.com/responsively-org/responsively-app/assets/87022870/92037763-d2aa-4322-b109-cf05f553b8c5)

This switch in the kebab menu of the toolbar switches off javascript in the site (by destroying and rebuilding the webviews) and turns it back 'on' using the same method.

Though while testing out the feature, I found out that the screenshot feature was crashing with this error:
`Error while taking full screenshot Error: Error invoking remote method 'GUEST_VIEW_MANAGER_CALL': Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error.`

Investigating further, the screenshot function uses WebContents and executes javascript inside the webview and this **doesn't** get executed as the feature disables javascript in the webview. I tried changing the attribute webpreferences on the fly during the screenshot function execution, but the webview doesn't take that into account and only allows javascript on a rebuild of the webview.

Would like some additional help regarding this.

**Temporary probable solution**
We could disable screenshot-function while the Javascript remains off.

cc: @themohammadsa @jjavierdguezas

**Comment by manojVivek** [comment]
Yes, I agree to disable the screenshot button when JS is disabled. 
Also, please add some note on hover of the disabled screenshot button to indicate that it is because JS is disabled so that the users are not puzzled about why it is disabled.

**Comment by ParamBirje** [comment]
@manojVivek as discussed, the screenshot button is now disabled for as long as the javascript remains disabled. 

After going through a series of test runs, I found out that after selecting and deselecting the "Inspect Elements" (toggling on/off) button in the main toolbar...

![inspect-element-toggle-webview-crash](https://github.com/responsively-org/responsively-app/assets/87022870/16f9fb0a-12d9-4cde-a028-738610953ee7)

then changing app's view (to Manage Suites / Going into individual device view mode) and coming back to the standard multiple layout view causes this error:

![webview-crash](https://github.com/responsively-org/responsively-app/assets/87022870/ab177f61-dc7c-4fda-88f0-7e70d12369bc)

I suspect this is probably happening because of the webviews getting rebuilt and the new webview reference is not getting updated somewhere. This happened to a couple of functions of the device specific toolbar but I got them fixed by updating them through useEffect. 

I'm trying to find where the current error is getting triggered.

cc: @themohammadsa

**Comment by Dailin521** [comment]
I checked the current `desktop-app` implementation and the conflict surface looks clear. I’m taking a narrow pass on this issue and will keep the change limited to the existing device preview/webview paths before opening a focused PR.
